python/hsml/engine/model_engine.py

```diff
@@ -224,30 +263,45 @@ def save(self, model_instance, model_path, await_registration=480):
                 pbar.set_description("%s" % step["desc"])
                 if step["id"] == 0:
                     # Create folders
-                    self._engine.mkdir(model_instance)
+                    self._engine.mkdir(model_instance.version_path)
                 if step["id"] == 1:
+
+                    def update_upload_progress(n_dirs=0, n_files=0):
```

Comment:
can this function `update_upload_progress` be a class level function, instead of passing it as arguments to other functions ?

Reply:
This function is specific for the step two, and doesn't apply to the other steps. I think it's better to have it embedded in this case.